### PR TITLE
fixed #12176 :Add approxEquals method for Mat3/Mat4/Quaternion/Vec2/Vec3/Vec4.

### LINKED ISSUE #12176: float value comparison needs approximate version in Math library
`operator==` should be absolute equal, add new comparison function like `approxEquals`.
For example:

```c++
inline bool Vec4::operator==(const Vec4& v) const
{
    return x==v.x && y==v.y && z==v.z && w==v.w;
}

inline bool Vec4::approxEquals(const Vec4& v) const {
    return math::isEqualF(x, v.x)
        && math::isEqualF(y, v.y)
        && math::isEqualF(z, v.z)
        && math::isEqualF(w, v.w);
}

```

Quaternion, Vec3, Mat3, Mat4 also needs to implement this.

### PR DESCRIPTION
**Pull request by dumganhar**
Re: #12176

### Changelog

* fixed #12176 :Add approxEquals method for Mat3/Mat4/Quaternion/Vec2/Vec3/Vec4.

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: native/cocos/math/Mat4.h [hunk 1]
```diff
@@ -918,6 +917,16 @@ class CC_DLL Mat4 {
...
 
+    /**
+     * Determines if this matrix is approximately equal to the given matrix.
+     */
+    inline bool approxEquals(const Mat4& v, float precision = CC_FLOAT_CMP_PRECISION) const {
+        return math::isEqualF(m[0], v.m[0], precision) && math::isEqualF(m[1], v.m[1], precision) && math::isEqualF(m[2], v.m[2], precision) && math::isEqualF(m[3], v.m[3], precision)
+            && math::isEqualF(m[4], v.m[4], precision) && math::isEqualF(m[5], v.m[5], precision) && math::isEqualF(m[6], v.m[6], precision) && math::isEqualF(m[7], v.m[7], precision)
+            && math::isEqualF(m[8], v.m[8], precision) && math::isEqualF(m[9], v.m[9], precision) && math::isEqualF(m[10], v.m[10], precision) && math::isEqualF(m[11], v.m[11], precision)
+            && math::isEqualF(m[12], v.m[12], precision) && math::isEqualF(m[13], v.m[13], precision) && math::isEqualF(m[14], v.m[14], precision) && math::isEqualF(m[15], v.m[15], precision);
+    }
```
Comment (PatriceJiang) [inline native/cocos/math/Mat4.h]: Is this block too large for a inline function?

File: native/tests/unit-test/src/ core_scenegraph_nodeeventprocessor.cpp [hunk 2]
```diff
@@ -21,20 +21,21 @@ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
```
Comment (PatriceJiang) [inline native/tests/unit-test/src/ core_scenegraph_nodeeventprocessor.cpp:21]: The file name contains a space?

File: native/cocos/math/Quaternion.h [hunk 3]
```diff
@@ -393,6 +392,13 @@ class CC_DLL Quaternion {
      */
     inline Quaternion &operator*=(const Quaternion &q);
 
+    /**
+     * Determines if this quaternion is approximately equal to the given quaternion.
+     */
+    inline bool approxEquals(const Quaternion &v, float precision = CC_FLOAT_CMP_PRECISION) const {
```
Comment (PatriceJiang) [inline native/cocos/math/Quaternion.h:398]: Need unit test in math_quaternion_test?

File: native/cocos/math/Mat4.h (hunk 1 above)
Comment (dumganhar) [reply native/cocos/math/Mat4.h]: Yep, Mat3 & Mat4 should not inline approxEquals function.

File: native/cocos/math/Quaternion.h (hunk 3 above)
Comment (dumganhar) [reply native/cocos/math/Quaternion.h:398]: 1

**Review by PatriceJiang — APPROVED** [review]

**Review by minggo — APPROVED** [review]

**Pull request merged**